frontend/src/metabase/core/components/TabLink/TabLink.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { MouseEvent, useCallback, useContext } from "react";
+import { LinkProps } from "metabase/core/components/Link";
+import { getTabId, getTabPanelId, TabContext } from "../Tab";
+import { TabLinkRoot, TabLabel } from "./TabLink.styled";
+
+export interface TabLinkProps<T> extends LinkProps {
+  value?: T;
+}
+
+function TabLink<T>({ value, children, onClick, ...props }: TabLinkProps<T>) {
+  const { value: selectedValue, idPrefix, onChange } = useContext(TabContext);
+  const tabId = getTabId(idPrefix, value);
+  const panelId = getTabPanelId(idPrefix, value);
+  const isSelected = value === selectedValue;
+
+  const handleClick = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      onClick?.(event);
+      onChange?.(value);
```

Comment:
I saw `onChange` is defined here:
https://github.com/metabase/metabase/blob/f879b7f0b4909fe02fd17a403916bd14b9a72074/frontend/src/metabase/core/components/Tab/TabContext.tsx#L7

But I could never find where `onChange` is being passed to the context. I suspect now we never receive `onChange` at all. If that's the case then we could change the context signature and remove this call.